<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Sophie's Professional Website</title>
  <link rel="stylesheet" href="styles.css">
  <script src="script.js" ></script>
</head>
<body>
  <header align="center">
    <h1> Hi there!  Nice to meet you :D </h1>
    <div align="center" class="intro-container">
      <div class="image-container">
        <img style="width:30vw; border-radius: 10%;" src="./images/happy_picture.jpg" alt="Your Picture">
      </div>
      <div class="intro-text-container">
        <p style="font-size:1.5vw"> <span class="orange-text"> Welcome to my personal website! </span> </p>
        <p>
          My name is Sophie Liu, and I am so glad you stopped by! I think you're here 
          to get to know me better (maybe I'll get to meet you someday too :)), 
          so I'll get right into it!
        </p>
        <p> 
          Currently, I study Computational Biology and Machine Learning at the School of Computer
          Science at Carnegie Mellon University. I am set to graduate in Spring 2024.
        </p>
      </div>
    </div>
  </header>

  <main>
    <section id="about-me">
      <div class="background-image-container"></div>
      <div class="content-container" style="background-color: rgba(255, 227, 196, 0.9); color:rgb(4, 39, 88)"">
        <h2 >A Little About My Professional Interests</h2>
        <p>  
          After I graduate, I will contribute fundamental improvements to the world of computational biology by creating 
          the tools that other scientists use to advance their research. 
        </p>
        <p>
          Some tasks that speak to my interests include: developing novel algorithms for high-throughput protein sequencing, 
          redefining how we simulate/predict protein dynamics and structure, discovering new underlying rules behind cell signal transduction through simulation, 
          and developing more accessible and intuitive <em>de novo</em> protein design pipelines.
        </p>
        <p>
          There is <em> so much </em> for me to learn about the field of computational biology, so I really enjoy problems that push me to think harder
          and teach me something new.
        </p>
      </div>
    </section>


    <section id="internships">
      <div class="intern-bg-container"></div>
      <div class="content-container" >
        <h2>Internships</h2>
        <p> Click on the cards below to learn more about my past internships during college! </p>
        <div class="card-container">
          <div class="card">
            <h3>Amgen 2023</h3>
            <p>Topics: AWS Cloud computing, Full Stack, IgFold, Protein Folding, Alphafold, RoseTTAFold Generative AI</p>
            <div class="hidden-content expanded-card-content">
              I am currently an intern at Amgen working on four projects. 
              <ul>
                <li>The first project involves creating an Amazon Machine Image for the Molecular modeling team.
                  By constructing the packer and ansible playbook I am learning a lot about
                  <span class="orange-text">AWS and cloud infrastructure and software configuration</span> 
                  as I try to add all the necessary add-ons to the AMI.        
                </li>
                <li>The second project involves more full stack responsibilties to make an internal 
                  webapp that allows users to submit arxiv papers, ask questions about the paper,
                   and recieve technically detailed and accurate answers. Doing this I have become 
                   proficient with <span class="orange-text">LangChain</span>, a framework for development
                   using language models,<span class="orange-text"> OpenAI api, React, Node.js, Flask,</span> and 
                   <span class="orange-text"> website deployment on AWS. </span>
                </li>
                <li>The third task pertains to <span class="orange-text">setting up a pipeline that allows scientists to predict the structure of 
                  antibodies</span>. My task was to take state of the start antibody folding scripts and write the necessary shell, 
                  python, and dockerfiles to create a scalable pipeline for folding single or multiple antibodies.
                </li>
                <li>
                  My final project involves creating a standard test that the company can use to benchmark the performance
                  of protein folding tools existing or new. With this procedure, the company can carefully evaluate newly released  
                  protein folding resources to see if they are worth incorporating. <span class="orange-text">This task effectively sets the stage for all downstream molecular
                  simulations done on potential drug targets.</span>
                </li>
              </ul>  
            </div>
          </div>
          <div class="card">
              <h3>Peptilogics 2022</h3> 
              <p>Topics: Peptide Folding, PyRosetta, Rosetta, Ab Initio Folding</p>
              <div class="hidden-content expanded-card-content">
                During the summer going into my junior year, I had the privilege of working at Peptilogics, a 
                company focused on peptide-based theraputics.
                <ul>
                  <li>The first project was to take in custom information encoded on a PDB file, and display it in the
                    log of a molecular visualization program called ChimeraX. The code was written in python using the 
                    chimerax.core.commands library, and enabled simulation scientists to have a cleaner view of their desired
                    data from an ensemble PDB. 
                  </li>
                  <li>The second project also involved adding commands to ChimeraX. My task was to create a command that allows
                    scientists to color PDB residues based on custom attributes besides b-factors like Rosetta Energy Score. </span>
                  </li>
                  <li>The third task was to create a peptide folding pipeline for the company using the <em>ab initio</em> script on 
                    Rosetta. I learned how to use the pyRosetta python library, Rosetta suite, python, and bash scripting to make 
                    pipelines to fold peptides in batch. To measure our performance, we used the RMSD and pandas, numpy toolkits to
                    compare against industry standards Alphafold and PEPFOLD. My custom parameters and procedure was able to outperform
                    both programs in the use case of peptides, and now is now being used by the company in their drug discovery pipeline.
                  </li>
                </ul> 
                <img src="./images/peptilogics.jpeg" alt="Peptilogics Picture">
              </div>
          </div>
        </div>
      </div>
      <div class="expanded-container">
        <div class="expanded-card"></div>
      </div> 
    </section>
    
    <section id="research">
      <div class="research-bg-container"></div>
      <div class="content-container">
        <h2>Research</h2>
        <p> Click on the cards below to learn more about my past research experiences! </p>
        <div class="card-container">
          <div class="card">
              <h3>Bahar Lab (Undergrad)</h3>
              <p>Characterizing GPCR Binding Sites </p>
              <p>Topics: GPCRs, Protein Dynamics, pandas, numpy, shell scripting </p>
              <div class="hidden-content  expanded-card-content">
                Since March 2021 of my freshman year, I have had the privilege of studying
                protein dynamics with my mentor Dr. Pemra Doruker and Dr. Burak Kaynak under Dr. Ivet Bahar.
                <ul>
                  <li> My project focuses using our in-house ESSA library to predict top binding sites for GPCRs
                    by measuring the impact of ligand binding on protein dynamics for each potential site.
                  </li>
                  <li> At the Bahar Lab, I have been able ot use a lot of ESSA, python, pandas, and numpy to 
                    handle large amounts of data efficiently. In addition, I use bash scripting to 
                    automate simulations and experimentation to circumvent manual data annotation. </span>
                  </li>
                </ul> 
                <center>
                <img src="./images/gpcr.png" style="height:200px; width:auto;" alt="GPCR Picture">
                <img src="./images/murdoch.jpeg" style="height:200px; width:auto;" alt="Murdoch Picture">
                </center>
              </div>
          </div>
        
          <div class="card">
            <h3>Kaeberlein Lab (HS)</h3>
            <p> Introduction to <em>C. elegans</em>, first academic lab position</p>
            <p>Topics: Alzheimer's disease, analyzing C. elegans thrashing patterns</p>
            <div class="hidden-content expanded-card-content">
              <p>
                Between my junior and senior year of high school, I sought out my first university lab position at the  Kaeberlein Lab from the 
                University of Washington Department of Lab Medicine and Pathology. The work was mostly <span class = "orange-text"> focused on wetlab technique and manual data 
                annotation. </span>
              </p>
              <p>
                At the Kaeberlein lab, I was tasked with:
              </p>
              <ul>
                <li>Maintaining strains of <em>C. elegans</em> that were models for Alzheimer's, Parkinson's, and more</li>
                <li>Setting up parallel experiments on our <span class="orange-text">Wormbot</span> using linux shell commands</li>
                <li>Manually analyzing the data that was collected automatically (I hadn't been introduced to the openCV python library yet)</li>
                <li>A side project to sort and organization 10,000+ entries of data using python/pandas (I was but a beginner coder at the time)</li>
              </ul>
              <p>The data annotation and wetlab work required a lot more patience and lab technique rather than computational thinking, but the hours of manual
                data annotation taught me a lot about perserverance and hard work. I am very grateful that my contributions landed me a place as a co-author on 
                the paper debuting Wormbot, our automated robot for running <em>C. elegans</em> experiments.
              </p>
              <p>
                <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6925079/" style="text-decoration: none; color: inherit;">
                  The full paper can be found here</a>.
              </p>
              <center><img src="images/wormbot.gif" style="height:200px; width:auto" alt="Wormbot Gif"></center>
            </div>
          </div>
        
          <div class="card">
            <h3>Inst. for Systems Biology (HS)</h3>
            <p>"BAMage: Bacteria, Antibiotics, Microbiome" (Please pardon HS Sophie)</p>
            <p> Topics: Intro. Python, Biological Modeling, Microbiome, Antibiotics, Evolution </p>
            <div class="hidden-content expanded-card-content">
              <p>In my junior year of high school, I applied to and was accepted to the Institute Systems Biology (ISB)
                Internship program which was supering exciting for me since the founder is Dr. Leroy Hood, a scientist who
                I've looked up to for a while. The internship program was focused on merging STEM and the arts together, so 
                all intern projects had to convey some scientific concept through a media of our choice.
              </p>
              <p>
                I partnered with a friend of mine I met before the program started. <span class ="orange-text"> We made two animations in python: one
                showcased the never ending cycle between antibiotic use and antibiotic resistance, and then the other animation focuses on the use of
                antibiotics on the microbiome. </span> 
                <a href="https://see.isbscience.org/steam2019/sophie-2/" style="text-decoration: none; color: inherit;">
                  Details about the implementation and project can be found here
                </a>.
              </p>
              <p>
                This was my first time coding at all (simultaneously with my experience at the Kaeberlein lab), but I had a lot of
                fun in the process! I definitely knew that computational biology was something I would be interested in pursuing.
    
              </p>
              <center><img src="./images/microbiome.gif" style="height:200px; width:auto;" alt="Microbiome Animation"></center>
            </div>
          </div>

          <div class="card">
            <h3>iGEM 2019 (HS)</h3>
            <p>Spatially Controlling attachment of Functional Proteins on 
              Bacterial Cellulose using Optogenetics</p>
            <p>
              Topics: Synthetic Biology, Student led research, Biological Modeling using ODEs and PDes
            </p>
            <div class="hidden-content expanded-card-content">
              <p>
                In high school, I was a student leader for our local high schoool iGEM (International Genetically
                Engineered Machine competition) team.  
              </p>
              <p>
                For our 2019 project, Bacto-Basics, we focused on precise, light-based control of bacterial cellulose (BC) 
                functionalization using optogenetics. Our team accomplished the following:
                
                Finally, we engaged the community by providing a mini iGEM experience 
                for students and sought public feedback. 
              </p>
              <ul>
                <li>
                  Engineered E. coli to attach fusion proteins to BC via cellulose binding domains (CBDs) when stimulated by red and blue light sensors. 
                </li>
                <li>
                  Designed a bioreactor with optimal conditions for K. rhaeticus growth and BC production using 
                Raspberry Pi-controlled sensors.
                </li>
                <li>
                  Researched and discussed gene expression modeling using ODEs and PDEs and the Tabor Model 
                </li>
                <li>
                  Prepared a device for light intensity control and data collection experiments.
                </li>
              </ul>
              <a href="https://2019.igem.org/Team:SoundBio/Design" style="text-decoration: none; color: inherit;">
                More information about our project can be found here.
              </a> 
              <center>
                <img src="./images/2019igem.jpeg" style="height:200px; width:auto;" alt="Team Picture 2019">
              </center>
            </div>
          </div>

          <div class="card">
            <h3>iGEM 2018 (HS)</h3>
            <p>A Synthetic Biology Alternative to the LAL Endotoxin Detection Assay</p>
            <p>
              Topics: Synthetic Biology, Student-led research
            </p>
            <div class="hidden-content expanded-card-content">
              <ul>
                <li>
                  Led project to recreate Factor C, the first protein in the LAL 
                  cascade, using Bacillus subtilis for protein synthesis.
                </li>
                <li>
                  Developed an alternative method for detecting cleaved Factor C,
                   bypassing intermediate steps and utilizing a visual assay.
                </li>
                <li>
                  Codon-optimized Factor C gene and split it into fragments, 
                  ligated them into pSB1C3 plasmid, and transformed E. coli for replication.
                </li>
                <li>
                  Conducted His-tagged protein purification of Factor C and verified 
                  its integration into Bacillus subtilis genome.
                </li>
                <li>
                  Led project for the 2018 InterLab study, analyzing trends in 
                  absorbance, fluorescence, and concentration of different promotors and regulators
                </li>
              </ul>
              <center>
                <img src="./images/2018igem.jpeg" style="height:200px; width:auto;" alt="Team Picture 2018">
              </center>
            </div>
          </div>
        </div>
      </div>
      <div class="expanded-container">
        <div class="expanded-card"></div>
      </div> 
    </section>

    <section id="hackathons">
      <div class="hackathon-container"></div>
      <div class="content-container">
        <h2>Hackathon Projects</h2>
        <div class="card-container">
            <div class="card">
              <h3>Tartan Hacks 2023</h3>
              <p>
                Title: Anwei
              </p>
              <p>
                Awards: Design for America (DFA) Prize for Best Social Impact Project
              </p>
              <div class="hidden-content expanded-card-content">
                <p><span class = "orange-text">Tartan Hacks is the biggest hackathon in Pittsburgh hosted by CMU each year.</span> I wanted my project 
                  to address the steep price of therapy, so I designed a prototype AI therapist powered by an OpenAI 
                  model I fine tuned myself on therapist transcripts.</p>
                <p>The final product was a seamless website that allowed users to: </p>
                <ul>
                  <li> 
                    Fill out exercises to help them process their thoughts, anxieties, and worries
                  </li>
                  <li> 
                    Chat with a therapist as if they were texting them
                  </li>
                  <li>
                    Sometimes, we need a face to face interaction though, so I made a prototype animated 
                    therapist that you can speak to as if it was a video call
                  </li>
                </ul>
                <p>I was working by myself for this hackathon, so I was pretty thrilled when my project won the best social impact prize :)</p>
                <p>This project was a great exercise to bolster my full stack development skills. In addition it was the first time
                  I worked with the OpenAI api, text-to-speech packages, and also my first time putting my random animation knowledge to use.
                </p>
                <p>
                  Fun fact: Anwei in Chinese, means "to comfort".
                </p>
              </div>
            </div>

            <div class="card">
              <h3>Tartan Hacks 2022</h3>
              <p>
                Title: Piano Translator
              </p>
              <p>
                Awards: Finalist in Overall Competition, Third Place
              </p>
              <div class="hidden-content">
                <p>
                  <span class = "orange-text">Tartan Hacks is the biggest hackathon in Pittsburgh hosted by CMU each year.</span>
                  During this time, I was starting to get back into piano practice, but all of the songs I wanted to play did not have
                  sheet music. However they did have synthesia videos, which is just a video of a keyboard at the bottom with notes falling
                  down the screen DDR style/Tap-tap revolution style. My teammate and I sought to turn these videos into sheet music 
                  music computer vision. 
                </p>
                <p>
                  <ul>
                    <li>
                      This project leveraged full stack development skills to provide a platform which users can use the tool.
                    </li>
                    <li>
                      It also required a lot of algorithmic creativity as we had to think of ways to detect notes on a keyboard, 
                      what they were, when they were played, interpret musical rhythms from the number of seconds a note was played.
                      Not to mention we had to figure out how to separate left and right hand notes, and piece them together.
                    </li>
                    <li>
                      Finally, we had to take all that information and translate into Music XML format which is how we decided to encode
                      the sheet music information.
                    </li>
                  </ul>
                </p>
                <p>
                  We presented this at the final round of TartanHacks 2022, and you can see our demo at the timestamp 51:36. Please note
                  I was indeed extremely sleep deprived during the presentation, so my coherent speaking capability was somewhat hindered:
                </p>
                <a href = "https://cmu.zoom.us/rec/play/ZHgLUYqFncO591o4LfxAtRKah6tQ8Kk3vivcfgDQcWhVjGMgR3vDQcEAFCEFESAFhwLjoFZtZbRW3dJ4.yXjWasTwnetU8O5J?continueMode=true&_x_zm_rtaid=yiaY8vd4RHi888MKYeYI3g.1662528341149.8a1376cfc4847dc3442995c73dc94516&_x_zm_rhtaid=655"> Presentation Link </a>
                <a href = "https://github.com/sophie006liu/piano_translator"> Github Repo </a>

              </div>
            </div>
            <div class="card">
              <h3>CMU Hack 112</h3>
              <p>
                Title: Poltical Engagement
              </p>
              <p>
                Awards: Most Social Impact, Runner Up: Most Interdisciplinary Project
              </p>
              <div class="hidden-content">
                <p>
                  During my first college semester, I entered my first ever hackathon where our team, HackeyMackey, developed a tools to increase
                  voter activity. Our website had 2 key components:
                  <ul>
                    <li>
                      A quiz to determine your political alignment
                    </li>
                    <li>
                      An automated system that takes issues you care about drafts an email with supporting 
                      evidence to send to your local representative (and we'll find their email for you ;)).
                    </li>
                    <li>
                      Note this was pre-chatgpt so the emails were created algorithmically using web scraping tools
                      and our own text analysis script.
                    </li>
                  </ul>
                </p>
                <p>The github reposity can be found </p> <a href="https://github.com/michyzhu/hackeymackey112">here.</a>
              </div>
            </div>
        </div>
      </div>
    </section>

    <section id="teaching" style="background-color:#cce1f5;">
      <div class="content-container">
        <h2>Teaching Experience</h2>
        <p>
          Since high school, I found gratification in seeing my peers find joy in learning. Before college,
          I tutored students in subjects from biology, chemistry, algebra, calculus, english, and even debate!
        </p>
        <p>
          I will be having my first experience in August 2023 as a teacher's assistant in college for the course
          15-451/651 Algorithm Design and Analysis. We will be covering topics like amortized analysis for potential functions,
          linear programming, dynamic programming, min-cut max-flow, hashing, segtrees, streaming algorithms, and computaitonal geometry!
        </p>
      </div>
    </section>

    <section id="coursework" style="background-color:#cce1f5;">
      <div class="content-container">
        <h2>Coursework</h2>
        <p>
          I'm currently pursuing a minor in Machine Learning. Here are some courses I've taken/plan to take:
        </p> 
        <p>**Taken - Introduction to Machine Learning (10-315):**</p>

        <div style="display: flex; justify-content: space-between;">
          <div style="flex: 50%; width: 45%;">
            <ul>
              <li>Supervised Learning: Linear Regression</li>
              <li>Supervised Learning: Logistic Regression</li>
              <li>Supervised Learning: Neural Networks</li>
              <li>Supervised Learning: Naive Bayes</li>
              <li>Supervised Learning: Support Vector Machines</li>
            </ul>
          </div>
          <div style="flex: 50%; width: 45%;">
            <ul>
              <li>Unsupervised Learning: K-Means Clustering</li>
              <li>Unsupervised Learning: Gaussian Mixture Models</li>
              <li>Unsupervised Learning: Dimensionality Reduction</li>
              <li>Probability and Bayes Rule in Machine Learning</li>
              <li>Model Selection, Overfitting, and Regularization Techniques</li>
            </ul>
          </div>
        </div>
        <p>**Taken - Deep Reinforcement Learning and Control (10-405):**</p>
        <div style="display: flex; justify-content: space-between;">
          <div style="flex: 50%; width: 45%;">
            <ul>
              <li>Evolutionary Methods for Policy Search</li>
              <li>Markov Decision Processes (MDPs)</li>
              <li>Value Iteration and Policy Iteration</li>
              <li>Monte Carlo Learning</li>
              <li>Temporal Difference Learning</li>
            </ul>
          </div>
          <div style="flex: 50%; width: 45%;">
            <ul>
              <li>Function Approximation in Prediction and Control</li>
              <li>Model-Based Reinforcement Learning (MBRL)</li>
              <li>Monte Carlo Tree Search (MCTS)</li>
              <li>Actor-Critic Methods</li>
              <li>Policy Optimization Algorithms (e.g., PPO, TRPO)</li>
            </ul>
          </div>
        </div>
      <p>**Taking Fall 2023 - Intermediate Deep Learning (10-417):</p>

      <div style="display: flex; justify-content: space-between;">
        <div style="flex: 50%; width: 45%;">
          <ul>
            <li>Linear Algebra for Deep Learning</li>
            <li>Probability Distributions and Rules</li>
            <li>Regression and Classification Algorithms</li>
            <li>Feedforward Neural Networks and Backpropagation</li>
            <li>Optimization Techniques: Dropout and Batch-Normalization</li>
          </ul>
        </div>
        <div style="flex: 50%; width: 45%;">
          <ul>
            <li>Convolutional Neural Networks (CNNs) for Computer Vision</li>
            <li>Autoencoders and Sparse Coding Models</li>
            <li>Deep Generative Models: DBNs and DBMs</li>
            <li>Generative Adversarial Networks (GANs)</li>
            <li>Sequence Modeling: Recurrent Neural Networks (RNNs) and Transformers</li>
          </ul>
        </div>
      </div>
    </div>
    </section>

    <section id="hobbies" style="background-color:#fefbe0;">
      <div class="content-container">
        <h2>Hobbies</h2>
        <p>
          In my free time, I like to play frisbee and rock climb with my friends! 
        </p>
        <p>
          Music is a huge part of my life, and I love relaxing by noodling around on the keyboard. 
          In addition, I have recently picked up acoustic and electric guitar, and I enjoy jamming out
          to my favorite songs.
        </p>
        <p>
          I also like drawing, dancing, running, making friends, baking, and cooking :) I've 
          included some of my drawings/paintings below:
        </p>
        <div class="card-container">
          <div class="card">
            <img src="./images/building_over_water.jpg" alt="Your Picture">
          </div>
      
          <div class="card">
            <img src="./images/flow.png" alt="Your Picture">
          </div>
      
          <div class="card">
            <img src="./images/flower.jpeg" alt="Your Picture">
          </div>
      
          <div class="card">
            <img src="./images/cloud.png" alt="Your Picture">
          </div>
          
          <div class="card">
            <img src="./images/champs.jpeg" alt="Your Picture">
          </div>
          <div class="card">
            <img src="./images/watercolor_flowers.jpeg" alt="Your Picture">
          </div>
        </div>
      </br>
      </div>
      <div class="expanded-container">
        <div class="expanded-card"></div>
      </div> 
    </section>
  </main>


  <footer>
    <p>&copy; 2023 Sophie Liu. </p>
  </footer>
</body>
</html>
